client/src/components/dashboard.js

```diff
@@ -4,11 +4,27 @@ import { channels } from '../services/api';
 
 import './dashboard.css';
 
-function Dashboard() {
+function Dashboard(props) {
   const [fetchedChannels, setChannels] = useState(undefined);
   const [searchKeyword, setKeyword] = useState('');
   const [error, setError] = useState(null);
 
+  function isChannelMember(channel) {
+    const userId = props.user.id;
+    return channel.members.includes(userId);
+  }
+
+  function isChannelOwner(channel) {
+    const userId = props.user.id;
+    return channel.ownerId === userId;
+  }
+
+  async function joinChannel(event) {
+    const channelId = event.target.id;
+    const userId = props.user.id;
+    const channel = await channels.joinChannel(channelId, userId);
```

Comment:
We should return the result of the api fetch to be used elsewhere. In this case, for redirecting to its route